running/ReplacingDocker/podman.sh

```diff
@@ -0,0 +1,311 @@
+#!/bin/sh
+
+# podmah.sh demo script.
+# This script will demonstrate a lot of the features of podman, concentrating
+# on the security features.
+
+buildah_image() {
+    sudo podman images  | grep -q -w buildah-ctr
+    if [[ $? != 0 ]]; then
+	sudo podman build -t buildah-ctr -f Dockerfile.buildah .
+    fi
+}
+
+setup() {
+    rpm -q podman buildah audit >/dev/null
+    if [[ $? != 0 ]]; then
+	echo $0 requires the podman, buildah and audit packages be installed
+	exit 1
+    fi
+    command -v docker > /dev/null
+    if [[ $? != 0 ]]; then
+	echo $0 requires the docker package be installed
+	exit 1
+    fi
+    sudo cp /usr/share/doc/audit/rules/10-base-config.rules /etc/audit/rules.d/audit.rules
+    sudo augenrules --load > /dev/null
+    sudo systemctl restart auditd 2> /dev/null
+    sudo systemctl restart docker
+    clear
+}
+version() {
+    # Podman inside a container
+    read -p "sudo podman version"
+    echo ""
+    sudo podman version
+    echo ""
+
+    # Podman inside a container
+    read -p "sudo podman info"
+    echo ""
+    sudo podman info
+}
+
+rootless() {
+    # Rootless podman
+    read -p "Podman as rootless"
+    echo ""
+
+    read -p "--> podman pull alpine"
+    podman pull alpine
+    echo ""
+
+    read -p "--> podman run --net=host --rm alpine ls"
+    podman run --net=host alpine ls
+    echo ""
+
+    echo "Show Non Privileged containers"
+    read -p "--> podman ps -a"
+    podman ps -a
+    echo ""
+
+    echo "Show Non Privileged images"
+    read -p "--> podman images"
+    podman images
+    echo ""
+
+    echo "Show Privileged images"
+    read -p "--> sudo podman images"
+    sudo podman images
+    echo ""
+
+    read -p "--> clear"
+    clear
+}
+
+userns() {
+    echo "
+The demo will now unshare the usernamespace of a rootless container,
+using the 'buildah unshare' command. Explore your home directory to see
+what it looks like while in a user namespace. 'cat /proc/self/uid-map' and
+'cat /etc/subuid' to understand how 'buildah' created the user namespace.
+
```

Comment:
OK I tried the cat commands when the script put me into the command line and not much was shown in the two files you tell folks to look at?

```
Explore your home directory to see
what it looks like while in a user namespace. 'cat /proc/self/uid-map' and
'cat /etc/subuid' to understand how 'buildah' created the user namespace.

Type 'exit' to continue running the demo.

--> buildah unshare
[root@localhost ~]# cat /proc/self/uid-map
cat: /proc/self/uid-map: No such file or directory
[root@localhost ~]# ls
anaconda-ks.cfg  Dockerfile.buildah  go  myvol  podman.sh  tsweeney
[root@localhost ~]# cat /etc/subuid #no values returned
```

Reply:
/etc/subuid is on the host not in the container
I can add the cat to the script.

Reply:
Should cat /proc/self/uid_map, Fixed in next version.